libbpfgo.h

```diff
@@ -131,6 +131,47 @@ int bpf_prog_detach_cgroup_legacy(
   return syscall(__NR_bpf, BPF_PROG_DETACH, &attr, sizeof(attr));
 }
 
+// https://lore.kernel.org/all/20220824233117.1312810-2-haoluo@google.com/
+#ifndef bpf_cgroup_iter_order
+enum bpf_cgroup_iter_order {
+  BPF_ITER_ORDER_UNSPEC = 0,
+  BPF_ITER_SELF_ONLY,        /* process only a single object. */
+  BPF_ITER_DESCENDANTS_PRE,  /* walk descendants in pre-order. */
+  BPF_ITER_DESCENDANTS_POST, /* walk descendants in post-order. */
+  BPF_ITER_ANCESTORS_UP,     /* walk ancestors upward. */
+};
+#define BPF_ITER_LINK_INFO_NO_CGROUP
+#endif
+
+// https://lore.kernel.org/bpf/20220926184957.208194-2-kuifeng@fb.com/
+#ifndef bpf_iter_task_type
+#define BPF_ITER_LINK_INFO_NO_TASK
+#endif
```

Comment:
What do we have to assume from these defines? I ask since `bpf_prog_attach_iter()` is setting up opts to `bpf_program__attach_iter()`. Which libbpf version started to use opts like this?

Reply:
@mozillazg please take a look, https://github.com/mozillazg/libbpfgo/pull/8

run make clean and make install again, you'll have the uapi/linux headers in output to be used.

Reply:
We no longer need to add these defines after modifying the Makefile to use the libbpf vendored linux/bpf.h .

Reply:
@mozillazg I think we'll first handle https://github.com/aquasecurity/libbpfgo/issues/242, so this whole work (using vendor headers) won't be necessary.

I'll tackle #242 asap and will let you know.

Reply:
@mozillazg reviewing this in hindsight while working on #288, I think we must use your last proposal, redefining types. libbpf curiously only defines them in the uapi headers and we can't mix bpf and uapi/bpf, since we'll face redefinitions.

So, it would be nice to define the missing types in a separate file like defs.h and include it in libbpfgo.h.

After doing that, please squash the commits of this PR.